java.base/src/main/java/java/lang/Class$_native.java

```diff
@@ -1,25 +1,17 @@
 package java.lang;
 
-import java.lang.reflect.Field;
+import org.qbicc.runtime.main.CompilerIntrinsics;
+import org.qbicc.runtime.main.VMHelpers;
+
+import static org.qbicc.runtime.CNative.*;
 
-// TODO: These are all stubs so we don't turn them into
-//       an UnsatisfiedLinkError before the interpreter can
-//       intercept them at build time.
-//       Eventually we need a real runtime implementation too...
 public final class Class$_native<T> {
     public boolean isAssignableFrom(Class<?> cls) {
-        throw new UnsupportedOperationException();
-    }
-
-    private Field[] getDeclaredFields0(boolean publicOnly) {
-        throw new UnsupportedOperationException();
-    }
-
-    public Class<?> getSuperclass() {
-        throw new UnsupportedOperationException();
-    }
-
-    public int getModifiers() {
-        throw new UnsupportedOperationException();
+        Object pun = this;
+        Class<?> thisAsClass = (Class<?>) pun;
```

Comment:
This is clumsy, but since java.lang.Class is final, I couldn't see a less clumsy way to do get javac to compile the code...happy to change if there is a better idea.

Reply:
This is basically the right idiom.  In other places I have just done a double-cast like `Thing self = (Thing) (Object) this`.

Maybe we can introduce a compile-time-checked method to do this in the future, something like `Thing self = safeCast(this, Thing.class)`.  IIRC GraalVM has something like that.